### PR TITLE
Make nat and nft rules consistent; improve rule ordering.

### PR DESCRIPTION
**Pull request by skuhl**
First, check if TTL indicates we should ignore packet (instead of
checking in multiple rules later). Also, nft method didn't do this at
all. Now, nft matches the behavior of nat.

Second, forward DNS traffic (we may need to intercept traffic to
localhost if a DNS server is running on localhost).

Third, ignore any local traffic packets. (Previously, we ignored local
traffic except DNS and then had the DNS rules). The nft method didn't
do this previously at all. It now matches the behavior of nat.

Lastly, list the subnets to redirect and/or exclude. This step is left
unchanged. Excluding the local port that we are listening on is
redundant with the third step, but should cause no harm.

In summary, this ordering simplifies the rules in nat and eliminates
differences that previously existed between nat and nft.

### CODE REVIEW

**Comment by skuhl** [comment]
If this gets merged, I'll submit another patch that depends on this one that adds IPv6 support to the nft method.

**Comment by brianmay** [comment]
Thanks!
